### ISSUE TITLE
conda python interactive command line SIGSEGV error

### ISSUE THREAD

**Issue by Ruibin-Liu** [fish-shell/fish-shell#10192]
- fish version: 3.7.0
- OS: Linux 6.2.0-39-generic 40~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Thu Nov 16 10:53:04 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
- Without third-party customizations: same behavior

I installed Anaconda version 23.11.0 for python 3.10.13. Python can be used as a command but the interactive command line interface has the following error:

```
Python 3.10.13 (main, Sep 11 2023, 13:44:35) [GCC 11.2.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
fish: Job 1, 'python' terminated by signal SIGSEGV (Address boundary error)
```

Three related observations:

1. I also tried Miniconda and the latest Anaconda (regarding to the Python version). They also had the same problem.

2. No issue if using Bash.

3. No issue if using Python installed through apt.

**Comment by faho** [comment]
Have you tried talking to the conda people?

It's a crash in their tool, so they'll be more likely to help you.

What I can suggest is first of all looking at differences in the environment between both.

1. Check `type python` - if that's not the same you're running different interpreters
2. Check `env` in fish vs bash

**Comment by faho** [comment]
> I also tried Miniconda and the latest Anaconda (regarding to the Python version). They also had the same problem.


I can not reproduce this after installing miniconda 23.5.2 with python 3.11.4.

I run `/opt/miniconda3/bin/conda shell.fish activate base | source` and then `python` (or `python3`) and the interpreter starts.

**Comment by Ruibin-Liu** [comment]
@faho Thanks for checking! I have identified the problem by trying the `env` command in both `bash` and `fish`. 

The guy installed the system used a wrong locale which only `bash` complained whenever I switched to it (don't know why it did that though). And the conda python could not run in `bash` as well, so I thought that could cause the problem and thus I updated the related environmental variables in `bash` in several places like `~/.profile`, and then conda python could run. Using `env` showed the related variables like `LANG` and `LC_CALL` were still the wrong one in `fish`. I changed them in `.config/fish/config.fish` to the correct one and now conda python can run in the interactive mode.
